src/riscv-integration.adoc

```diff
@@ -791,6 +805,23 @@ CHERI violations have the following order in priority:
 . Invalid address violation
 . Bounds violation (_Lowest_)
 
+[#mtval2,reftext="mtval2"]
+==== Machine Trap Value Register 2 (mtval2)
+
+The <<mtval2>> register is an MXLEN-bit read-write register, which is added as part of the
+Hypervisor extension. {cheri_base_ext_name} also requires the implementation of this CSR.
+
+When a CHERI fault is taken on any data memory access into into M-mode, <<mtval2>> is written
+with the MXLEN-bit address which caused the fault. This can be used together with the
+CAP_IDX field of <<mtval>> to diagnose the authorising capability which caused the
+exception.
+
+<<mtval2>> is set to zero for all PCC faults, and follows the standard rules for non-CHERI exceptions.
```

Comment:
By PCC faults, do you mean faults for checks where the authorising capability is PCC?

Reply:
`PCC fault` is already in the doc